dht/dht.go

```diff
@@ -618,13 +648,28 @@ func (s *Server) nodeByID(id string) *node {
 	return nil
 }
 
-func (s *Server) handleQuery(source dHTAddr, m Msg) {
+func (s *Server) handleQuery(source DHTAddr, m Msg) {
+	var (
+		newmsg *Msg
+		skip   bool
+	)
 	node := s.getNode(source, m.SenderID())
 	node.lastGotQuery = time.Now()
 	// Don't respond.
 	if s.config.Passive {
 		return
 	}
+	// Call any hooks that may apply
```

Comment:
I think the OnQuery hook should be called even if the server is set to passive. Move this up above the passive check?